### PR TITLE
Apply PHP 7.4 syntax and typed property

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 7.4, php 7.4 syntax can be applied with typed properties. 

- For `final class`, typed properties for protected is allowed if no inherit of parent or used by its final class parent.
- For `final class`, typed properties for protected can be changed to `private` if not used in parent or exists in parent class.
- Otherwise, update only private properties.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

**Review by gsteel — APPROVED** [review]

File: test/PHPUnit/Controller/MemoryLeakTest.php [hunk 1]
```diff
@@ -40,6 +40,6 @@ public function testMemoryConsumptionNotGrowing($null)
         $this->assertNull($null);
 
         // Test memory consumption is limited to 5 MB for 100 tests
-        $this->assertLessThan(5242880, memory_get_usage(true) - self::$memStart);
+        $this->assertLessThan(5_242_880, memory_get_usage(true) - self::$memStart);
```
Comment (froschdesign) [inline test/PHPUnit/Controller/MemoryLeakTest.php:43]: Since version 7.4, decimal numbers can contain underscores, but do we use them?

Comment (samsonasik) [reply test/PHPUnit/Controller/MemoryLeakTest.php:43]: we use them in other my PRs that were merged by @Ocramius

Comment (Ocramius) [reply test/PHPUnit/Controller/MemoryLeakTest.php:43]: It's really just a style choice. No big effect here, since we're in a test :)

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

File: test/PHPUnit/Controller/MemoryLeakTest.php (hunk 1 above)
Comment (froschdesign) [reply test/PHPUnit/Controller/MemoryLeakTest.php:43]: @Ocramius 
> It's really just a style choice.

Sure, but not defined, therefore my question. But I have no problem with that, only that we make you a reference and at the same time a standard. 😄

Comment (Ocramius) [reply test/PHPUnit/Controller/MemoryLeakTest.php:43]: I think it's just Rector's preset from @samsonasik: we could provide a CS rule for this in `laminas/laminas-coding-standard`, if we wanted to internalize it.

From the standpoint of this kind of patches, I'd say that it "doesn't matter", as in I prefer constant and gradual improvements, regardless if they introduce some CS inconsistencies that are in a gray zone :)

**Pull request merged**